test/handlers/code/backspace-inline.js

```diff
@@ -0,0 +1,11 @@
+/** @jsx h */
+
+export default input => input.backspace().backspace().backspace()
+
+export const input = <editor>
+    <p>Text<codeline>So<cursor/>me code</codeline></p>
+</editor>
+
+export const output = <editor>
+    <p>Tex<codeline><cursor/>me code</codeline><text/></p>
```

Comment:
I believe it's because just before the inline and just within the inline are two separate positions, and operations behave differently in them, so slate chooses the one which is more consistent with cursor's previous position, in this case just inside. Were this test for double delete two characters before the inline I think the cursor would be placed just before the inline.